src/services/UserService.ts

```diff
@@ -0,0 +1,22 @@
+import { UploadedPromotionsResponse } from '../types/api';
+import { Promotion } from '../types/promotion';
+import { isError } from '../utils/api';
+import Routes from '../utils/routes';
+
+class UserService {
+  public async getUploadedPromotions(userId: string): Promise<Promotion[]> {
+    const url = `${Routes.USERS}/${userId}/uploadedPromotions/`;
+    return fetch(url)
+      .then((response: Response) => response.json())
+      .then((response: UploadedPromotionsResponse) => {
+        if (isError<UploadedPromotionsResponse>(response)) {
+          return Promise.reject(response);
+        }
+        const { uploadedPromotions } = response;
+        return Promise.resolve(uploadedPromotions);
```

Comment:
Also, I think we should talk about how much detail we want to show in the uploaded and saved promotions page compared to the main section. 

I wasn't really too sure if for uploaded and saved promotions we want to return the discounts and schedules as well

Reply:
Ah good point, I didn't notice that. From the design, I just used the regular PromotionCard component that we display on the home page so I think it just uses schedules?

![Screen Shot 2021-02-21 at 1 50 23 PM](https://user-images.githubusercontent.com/44531733/108639965-1692fb80-744c-11eb-8f49-9b5a1b7eef0d.png)

Reply:
Oh I see, hmmm right now I noticed that for BE we return the user information and we also don't include the schedules or discounts. I think we can just return an array of promotions joined with the discounts/schedules.